synapse/events/third_party_rules.py

```diff
@@ -76,8 +76,7 @@ def __init__(self, hs):
                 public_room_manager=PublicRoomsManager(hs),
             )
 
-    @defer.inlineCallbacks
-    def check_event_allowed(self, event, context):
+    async def check_event_allowed(self, event, context):
```

Comment:
We needed to `async` this method to eventually call the check on whether a room is public. Luckily this method is already `await`ed on everywhere!